google/resource_compute_instance.go

```diff
@@ -1101,6 +1163,31 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 		d.SetPartial("scheduling")
 	}
 
+	if d.HasChange("status") {
+		var op *compute.Operation
+		expectedStatus := d.Get("status").(string)
+		if expectedStatus == "RUNNING" {
+			op, err = config.clientCompute.Instances.Start(project, zone, instance.Name).Do()
+			if err != nil {
+				return err
+			}
+		} else if expectedStatus == "TERMINATED" {
+			op, err = config.clientCompute.Instances.Stop(project, zone, instance.Name).Do()
+			if err != nil {
+				return err
+			}
+		} else {
```

Comment:
Since only `RUNNING` and `TERMINATED` status are allowed (validation when setting the argument), this code is not reachable but it is an additional precaution. Is it really needed?